### DISCUSSION TITLE
listDocuments gives 'missing scope' error

### DISCUSSION THREAD

**Discussion by hialvaro** [appwrite/appwrite#3021] [Q&A] [answered]
I am trying to listDocuments from a collection. Both authenticated users and non-authenticated users should be able to list (read) the documents from that collection. I am getting the following error: `User (role: guest) missing scope (account)`.

The code I am using to perform the request is:

```
try {
    isLoading.value = true;
    const user = await appwrite.account.get();
    const tempAwards = await appwrite.database.listDocuments(
      Server.collectionID as string
    );

    awards.value = tempAwards.documents
      ?.map((award) => ({
        ...award,
        owned: userOwnsAward(award, user),
      }))
      .sort((x, y) => (x.owned === y.owned ? 0 : x.owned ? -1 : 1));
    isLoading.value = false;
  } catch (error) {
    console.warn(error as string);
  }
});
```

Any ideas why I get this error? :/

**Comment by maeddin** [comment, marked as answer]
In your second line you call `account.get()`. That's the reason for the error.
If you want `user` to be nullable, you should catch this error and set `user` to `null` in the catch block.

**Reply by hialvaro** [reply]
This worked, thanks.

**Comment by hialvaro** [comment]
Also, I have one more question. Is there any way to listen for any session change? I mean, I want to know if the user logs in / logs out so I can change the layout. I found it can be done with supabase, but I can't find any info on how to do it with Appwrite:

```
supabase.auth.onAuthStateChange((_, session) => {
  store.methods.setUser(session);
  appReady.value = true;
});
```

**Reply by maeddin** [reply]
I don't think this is possible with the current version.

**Reply by maeddin** [reply]
I wrote something like that myself, since I wrote a wrapper for AppWrite anyway. When logging in, logging out and on certain flying errors you can update this value. But this is of course relatively cumbersome compared to Firebase and Supabase.

**Reply by hialvaro** [reply]
Hmm, I see. Is this planned for future versions?
